portalconta/src/main/java/br/com/coderbank/portalconta/services/ContaService.java

```diff
@@ -0,0 +1,67 @@
+package br.com.coderbank.portalconta.services;
+
+
+import br.com.coderbank.portalconta.dtos.requests.ContaFinanceiraRequestDTO;
+import br.com.coderbank.portalconta.dtos.responses.ContaFinanceiraResponseDTO;
+import br.com.coderbank.portalconta.dtos.responses.SaldoResponseDTO;
+import br.com.coderbank.portalconta.entities.Conta;
+import br.com.coderbank.portalconta.repositories.ContaRepository;
+import org.springframework.beans.BeanUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.util.UUID;
+
+@Service
+public class ContaService {
+
+    @Autowired
+    private ContaRepository contaRepository;
+
+    public ContaFinanceiraResponseDTO salvar(final ContaFinanceiraRequestDTO contaFinanceiraRequestDTO) {
+
+        var contaEntity = new Conta();
+
+
+       contaEntity.setNumeroAgencia(1);
+
+
+        contaEntity.setSaldo(BigDecimal.ZERO);
+
+
+        contaEntity.setNumeroConta(geraNumeroConta());
+
+            BeanUtils.copyProperties(contaFinanceiraRequestDTO, contaEntity);
+            contaRepository.save(contaEntity);
+
+            return new ContaFinanceiraResponseDTO(contaEntity.getIdConta(),
+                    contaEntity.getIdCliente(),
+                    contaEntity.getNumeroAgencia(),
+                    contaEntity.getNumeroConta(),
+                    contaEntity.getSaldo()
+
+
+            );
+
+
+    }
+    private Integer geraNumeroConta(){
+        var gerador = this.contaRepository.findAll().size();
+        return  gerador+1;
+
+
+
+    }
+
+    public SaldoResponseDTO obterSaldoPorIdCliente(UUID idCliente ) {
+        var saldo = contaRepository.findByIdCliente(idCliente).get().getSaldo();
```

Comment:
Se não existir uma conta para o idCliente fornecido, o método contaRepository.findByIdCliente(idCliente) retornará um Optional vazio. Quando você chama .get() em um Optional vazio, uma exceção NoSuchElementException será lançada.

Para evitar essa exceção, é uma boa prática verificar se o Optional contém um valor antes de chamar .get(). Aqui está um exemplo de como você pode fazer isso:
```

Optional<Conta> contaOptional = contaRepository.findByIdCliente(idCliente);
if (contaOptional.isPresent()) {
    double saldo = contaOptional.get().getSaldo();
    // faça algo com o saldo
}
```

agora se contaOptional.isPresent for false, precisas implementar uma exceção personalizada utilizando um exception handler para retornar uma mensagem de erro personalizada para quem chamou o endpoint